api/src/main/java/org/openmrs/module/reporting/report/service/db/HibernateReportDAO.java

```diff
@@ -214,7 +216,45 @@ public List<ReportRequest> getReportRequests(ReportDefinition reportDefinition,
 		}
 		return c.list();
 	}
-	
```

Comment:
Add a method here that returns an Integer called "getReportRequestCount" or something, that takes in your parameters and returns the rowcount projection only, then modify the below method to just return the List<ReportRequest> that match your parameters including the pageNumber and pageSize.

Reply:
Added the method, also removed the "pagination concept", and replaced with more general first result and max results - the page calculation will be added to reporting-rest.